### PR TITLE
ci: be selective about when to offer tmate

### PR DESCRIPTION
**Pull request by jsoref**
**Description of your changes:**

Be selective about when to offer tmate
* Unconditionally offer it when running from the rook org
* Offer it if the current repository has a `USE_TMATE` secret
* Offer it if the workflow is being re-run

w/o this change, tmate is run on all forks when builds fail.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@jsoref Thanks for the PR but here I'm trying to understand the advantage of this change because currently we already have
1. We always get tmate when running in rook as PR 
2. we get tmate when re-running the Ci test

>w/o this change, tmate is run on all forks when builds fail.
regarding this, why we'll want to get two tmate for the same test one from rook/rook pr and one from the local fork?

please correct me my understanding is not correct.

**Comment by jsoref** [comment]
The advantage is for everyone else when they trigger actions in their forks of this repository.

Right now those failures automatically trigger tmate and spend an hour burning GitHub Action cycles. During that time, those slots can't be used by any other GitHub workflows. By default accounts have 20 concurrent slots. When one runs out of slots, any jobs that need slots have to wait.

**Comment by travisn** [comment]
@jsoref If CI cycles are to be avoided, you can also add the `skip-ci` label to the Rook PRs, in that case the integration tests will all be skipped. Does that help? As @subhamkrai mentioned, when we're running PRs, we generally want to debug them if they fail.

**Comment by jsoref** [comment]
Yes, technically I can, but this won't change how you experience this in the rook repository - a PR into rook will still get tmate. It'd only impact forks.

A PR into another organization's fork of rook wouldn't get tmate by default.

My changes should eventually get CI, when they're sent to rook, but when they're PR'd into my fork (which has no tags), it doesn't make sense.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -68,8 +68,15 @@ jobs:
       working-directory: /Users/runner/go/src/github.com/rook/rook
       run: tests/scripts/validate_modified_files.sh gen-rbac
 
-    - name: setup tmate session for debugging
+    - name: consider debugging
       if: failure()
+      run: |
+        if [ "$GITHUB_REPOSITORY_OWNER" = "rook" ] || [ -n "${{ secrets.USE_WEBSHELL }}" ] || [ "$GITHUB_RUN_ATTEMPT" -gt 1 ]; then
```
Comment (travisn) [inline .github/workflows/build.yml:74]: Where would `USE_WEBSHELL` be set? Or if this is a var we're just defining, how about just calling it the `USE_TMATE` secret? And how about a description such as this?
```suggestion
        # Enable tmate only in the Rook fork, where the USE_TMATE secret is set in the repo, or if the action is re-run
        if [ "$GITHUB_REPOSITORY_OWNER" = "rook" ] || [ -n "${{ secrets.USE_TMATE }}" ] || [ "$GITHUB_RUN_ATTEMPT" -gt 1 ]; then
```

**Review by travisn — CHANGES_REQUESTED** [review]
Makes sense to restrict running against other forks, just one more question...

File: .github/workflows/build.yml (hunk 1 above)
Comment (jsoref) [reply .github/workflows/build.yml:74]: I was on the fence between `webshell` and `tmate`. I'm totally happy to take your changes.

https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository

Here's an example url:
https://github.com/check-spelling/rook/settings/secrets/actions

<details><summary>Before</summary>

image
</details>

<details><summary>Creating a secret</summary>

https://github.com/check-spelling/rook/settings/secrets/actions/new

image

</details>

<details><summary>With secret</summary>

https://github.com/check-spelling/rook/settings/secrets/actions

image

</details>

**Review by travisn — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**